kaos/utils/interval_utils.py

```diff
@@ -0,0 +1,72 @@
+""" Utility functions to manage TimeInterval objects"""
+
+from itertools import tee
+
+from ..tuples import TimeInterval
+
+
+def pairwise(iterable):
+    """s -> (s0,s1), (s1,s2), (s2, s3), ... , (sn, None)
+    from: https://docs.python.org/2/library/itertools.html
+    """
+    iterable.append(None)
+    a, b = tee(iterable)
+    next(b, None)
+    return zip(a, b)
+
+
+def fuse_neighbor_intervals(input_list, assume_sorted=False):
+    """Merges neighboring TimeIntervals that share the same start/end time.
+    example: [(0,100),(100,200),(300,400)] -> [(0,200),(300,400)]
+
+    Assumptions:
+        Intervals don't overlaps.
+        Maximum of 2 intervals share any one border.
+        For any interval, there is maximum of one neighbor.
+
+    Args:
+        input_list (list of TimeInterval): list of time intervals to be fused.
+        assume_sorted (boolean): Whether input list is sorted with respect to TimeInterval.start
+                                 values. defaults to False.
+    """
+    if not assume_sorted:
+        input_list.sort(key=lambda x: x.start)
+
+    output_list = []
+    skip_next = False
+    for i, j in pairwise(input_list):
```

Comment:
I think there is an issue here, what would happen for this input:
[(0,100),(100,200),(200,400)]?

Reply:
The input breaks Assumption 3. Basically, View cone can't generate such inputs (there is actually a pointless check for it at line 80 of view_cone.py) so I didn't bother.

Reply:
Added support for this anyways :)